### PR TITLE
[ENHANCE] Upgrade torch version to 1.13.1

### PR DESCRIPTION
**Pull request by goodsong81**
- Upgrade torch==1.13.1+cu116 & torchvision==1.14.1+cu116
- Upgrade MMCV frameworks & NNCF for torch compatibility
- [ISSUE] Torch is 'build' prerequisite due to `mpl` extension for seg (irrelevant to runtime prerequisite)
  - <- Resolved by https://github.com/openvinotoolkit/training_extensions/pull/1620 by @JihwanEom 
- [ISSUE] Errors in action task
  - <- Some resolved, some skipped by https://github.com/openvinotoolkit/training_extensions/pull/1629
- [ISSUE] Anomalib depending on lower version of pytorch
  - To be addressed in other PR

### CODE REVIEW

**Comment by goodsong81** [comment]
[E2E test results]
image

- Anomaly cls/det/seg: torch incompatible -> @samet-akcay would create a separate PR for anomalib upgrade
- Action cls/det: failing from export -> @jaegukhyun @JihwanEom Could you try? Please check the samples, too.
- Others: FQ mismatches -> to be handled other PRs from @AlexanderDokuchaev

File: requirements/openvino.txt [hunk 1]
```diff
@@ -2,9 +2,9 @@
 # OpenVINO Requirements.                                                      #
 # FIXME: nncf 2.3.0 has scipy version conflict
 # this is a temporary workaround to udpate nncf dependencies
-nncf @ git+https://github.com/cih9088/nncf@9b204857c11bc694c2ca6a62fc9a575c002a0a99#egg=nncf
+nncf==2.4.0
```
Comment (goodsong81) [inline requirements/openvino.txt:3]: @cih9088 You don't need to create a separate PR :)

File: otx/algorithms/common/adapters/mmcv/models/backbones/torchvision_backbones.py [hunk 2]
```diff
@@ -25,9 +25,9 @@
 
 from mmcv.cnn import build_activation_layer, build_norm_layer
 from torch import nn
+from torch.hub import load_state_dict_from_url
 from torch.nn.modules.batchnorm import _BatchNorm
 from torchvision import models
-from torchvision.models.utils import load_state_dict_from_url
```
Comment (goodsong81) [inline otx/algorithms/common/adapters/mmcv/models/backbones/torchvision_backbones.py:31]: @harimkang This function had been moved to Torch hub.

File: pyproject.toml [hunk 3]
```diff
@@ -8,7 +8,7 @@ requires = [
     "wheel",
     "Cython~=0.29.32",
     "numpy~=1.21.0",
-    "torch@https://download.pytorch.org/whl/lts/1.8/cu111/torch-1.8.2%2Bcu111-cp38-cp38-linux_x86_64.whl"
+    "torch@https://download.pytorch.org/whl/cu116/torch-1.13.1%2Bcu116-cp38-cp38-linux_x86_64.whl"
```
Comment (goodsong81) [inline pyproject.toml]: @JihwanEom I'd like to remove this item. This is only required for `mpl` torch extension build. Could you create a draft PR to replace the custom operation by default one (I don't remember what it is)?

File: requirements/openvino.txt (hunk 1 above)
Comment (cih9088) [reply requirements/openvino.txt:3]: Thank you! Could you remove the comment above please? 😊

File: requirements/openvino.txt [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
 onnx==1.10.1
 openmodelzoo-modelapi==2022.3.0
 openvino==2022.3.0
 openvino-dev[tensorflow2,mxnet,caffe]==2022.3.0
-mmdeploy @ git+https://git@github.com/open-mmlab/mmdeploy@v0.10.0
+# FIXME: Below is the only one left as git+https install
+# But, the latest one on PyPI is 0.2.0. Might need custom build & upload
+mmdeploy @ git+https://git@github.com/open-mmlab/mmdeploy@v0.12.0
```
Comment (samet-akcay) [inline requirements/openvino.txt:10]: Would it be an idea to contact the mm folks for this? Like create an issue on their repo? Or email them directly?

File: pyproject.toml (hunk 3 above)
Comment (JihwanEom) [reply pyproject.toml]: Created #1620, please refer this PR.

File: requirements/openvino.txt (hunk 4 above)
Comment (goodsong81) [reply requirements/openvino.txt:10]: Yeah, I also thought so now :) I'll contact them.

File: tox.ini [hunk 5]
```diff
@@ -21,38 +21,35 @@ deps =
...
-    mmcv-full @ https://download.openmmlab.com/mmcv/dist/cu111/torch1.8.0/mmcv_full-1.6.1-cp38-cp38-manylinux1_x86_64.whl
+    torch @ https://download.pytorch.org/whl/cu116/torch-1.13.1%2Bcu116-cp38-cp38-linux_x86_64.whl
+    torchvision @ https://download.pytorch.org/whl/cu116/torchvision-0.14.1%2Bcu116-cp38-cp38-linux_x86_64.whl
+    torchtext @ https://download.pytorch.org/whl/torchtext-0.14.1-cp38-cp38-linux_x86_64.whl
     coverage
     pytest-timeout
     pytest-mock
     -r{toxinidir}/requirements/base.txt
-    -r{toxinidir}/requirements/anomaly.txt
+    # -r{toxinidir}/requirements/anomaly.txt
```
Comment (goodsong81) [inline tox.ini:47]: @ashwinvaidya17 Temporarily disabled this requirement as anomalib has torch version conflict. In this PR, all anomaly tests are failing due to absence of anomalib. What do you think? Would you create a PR to this branch? Or, will it be better to merge this PR and you create an update PR after that?

File: requirements/openvino.txt (hunk 4 above)
Comment (goodsong81) [reply requirements/openvino.txt:10]: https://github.com/open-mmlab/mmdeploy/issues/1720

**Review by sungmanc — APPROVED** [review]

**Review by supersoob — APPROVED** [review]

**Pull request merged**